Assets/Scripts/Environment/Hangar/Battery/BatteryInteractable.cs

```diff
@@ -80,14 +84,26 @@ public override void OnTriggerEnter(Collider other)
         if (other.CompareTag(HangarConstants.BatteryExitColliderTag))
         {
             DropBattery();
-            HangarManager.BatterySpawnPositionManager.RespawnBattery(transform, collider);
+            HangarManager.BatterySpawnPositionManager.RespawnBattery(transform, Collider);
         }
     }
 
+
+    public void PlaceBatteryInSlot(Transform slot)

```

Comment:
I feel like this should be in ``BatterySlotPlacement``, as it deals with slots specifically. 

Then we don't have to pass the Transform in as an argument either. We already have the ``BatteryInteractable`` object reference in the SlotPlacement too. 

But if you think it's better here we can leave it.

Reply:
I get what you are saying, but PlaceBatteryInSlot those 2 things that are within the BatteryInteractable class: 
 PlayerIsHoldingABattery = false;
  gameObject.SetParent(HangarManager.BatteriesContainer);

So I think it is best to leave them there. I don't like the idea of BatterySlotPlacement having to do those 2 things as it is not its responsibility. I think it is best to hide to BatterySlotPlacement the details of what placeBattery() does. 

BatteryInteractable manages the batteries' movement. So I think it fits nicely within this class. 

Or maybe there is something I am not seeing?

Reply:
That's cool, you're right, it does more stuff than I realised tbh :D 

I think I just saw that the name contained 'Slot' and I thought it should be in there lol. 

Leave it where it is now, like you said.